server/api/public.py

```diff
@@ -42,15 +47,106 @@
 # Specifying election results in a query param could cause us to hit URL size limits
 @api.route("/public/sample-sizes", methods=["POST"])
 def public_compute_sample_sizes():
-    compute_sample_sizes_input = request.get_json()
-    validate(compute_sample_sizes_input, COMPUTE_SAMPLE_SIZES_INPUT_SCHEMA)
-    sample_sizes = {
-        "ballotComparison": {},
-        "ballotPolling": {},
-        "batchComparison": {},
+    contest_for_sampler = parse_compute_sample_sizes_input(request.get_json())
+
+    audit_types = ["ballotComparison", "ballotPolling", "batchComparison"]
+    sample_sizes = {audit_type: {} for audit_type in audit_types}
+    sample_size_functions = {
+        "ballotComparison": compute_ballot_comparison_sample_size,
+        "ballotPolling": compute_ballot_polling_sample_size,
+        "batchComparison": compute_batch_comparison_sample_size,
     }
-    for risk_limit_percentage in list(range(0, 21)):
-        sample_sizes["ballotComparison"][str(risk_limit_percentage)] = 0
-        sample_sizes["ballotPolling"][str(risk_limit_percentage)] = 0
-        sample_sizes["batchComparison"][str(risk_limit_percentage)] = 0
+    for audit_type in audit_types:
+        for risk_limit_percentage in list(range(0, 21)):
+            sample_size_function = sample_size_functions[audit_type]
+            sample_sizes[audit_type][str(risk_limit_percentage)] = sample_size_function(
```

Comment:
Why strings for risk limit percentages? I don't have a strong opinion, just would have expected numbers

Reply:
Yeah good q! Even had I used numbers here, when calling `jsonify`, the numerical keys would've automatically been converted to string keys anyway (since JSON keys have to be strings).

So I decided to be explicit in my conversion to strings instead of implicit. The latter could result in other engineers digging to figure out what would happen to the numerical keys

Reply:
Ah, makes sense